### PR TITLE
API Performance Improvements

### PR DESCRIPTION
**Pull request by R2ZER0**
This PR improves the performance of API OrgGrantsMade and OrgGrantsReceived about 100x, by adding denormalised convenience fields to Grant model with recipient/funder org_ids with database indexes, eliminating the need to join with Recipient / Funder tables each query.

Code to load grant data into the db has been updated to also add these new fields.

Existing Grants are updated using a 3-step migration:
1. The new fields are added nullable, default NULL.
2. Data migration runs SQL to populate the new fields in existing Grants.
3. The new fields are altered to be non-nullable, all newly created Grants must specify the fields.

Given the new fields are non-nullable, and both existing load management commands are updated, it should be very unlikely that new Grants are created without the new fields properly populated or get out of sync.

With the goal of improving performance, this PR also:
* Updates API queries to specifically ignore the default order_by on Publisher
* Updates OrgGrantsMade/Received to prefetch Publisher data for the Grants
* Adds db indexes on org_id/name to Entity tables
* Increases Grants per page from 60 to 100

### CODE REVIEW

File: datastore/api/org/api.py [hunk 1]
```diff
@@ -103,14 +108,15 @@ class OrganisationGrantsMadeView(generics.ListAPIView):
 
     def get_queryset(self):
         org_id = self.kwargs.get("org_id")
+        print(f"Get Queryset {org_id}")
```
Comment (michaelwood) [inline datastore/api/org/api.py]: Probably not needed anymore?

File: datastore/db/migrations/0021_update_grants_convenience_fields.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+UPDATE_GRANTS_SQL = """
+UPDATE db_grant AS g
+SET publisher_org_id = p.org_id,
+    recipient_org_ids = ARRAY(SELECT jsonb_array_elements_text(jsonb_path_query_array((g.data)->'recipientOrganization', '$[*].id'))),
+    funding_org_ids = ARRAY(SELECT jsonb_array_elements_text(jsonb_path_query_array((g.data)->'fundingOrganization', '$[*].id')))
+FROM db_publisher AS p
+WHERE g.publisher_id = p.id;
+"""
```
Comment (michaelwood) [inline datastore/db/migrations/0021_update_grants_convenience_fields.py:16]: Slightly nervous about running this on 7,606,239 grants.. perhaps we constrain it to `latest`?

Comment (R2ZER0) [reply datastore/db/migrations/0021_update_grants_convenience_fields.py:16]: On a test on my laptop using a snapshot of the database with ~8 million grants, this took less than 5 minutes.

File: datastore/api/org/api.py (hunk 1 above)
Comment (R2ZER0) [reply datastore/api/org/api.py]: Removed

**Review by michaelwood — APPROVED** [review]

**Pull request merged**